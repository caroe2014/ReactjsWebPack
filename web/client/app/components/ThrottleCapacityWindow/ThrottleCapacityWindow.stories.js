// Copyright Â© 2018-2019 Agilysys NV, LLC.  All Rights Reserved.  Confidential Information of Agilysys NV, LLC.

import React from 'react';
import { storiesOf } from '@storybook/react';
import ThrottleCapacityWindow from '../ThrottleCapacityWindow';

storiesOf('ThrottleCapacityWindow', module)
  .add('With Keep time slot button', () => {
    let capacityWindows = [
      '1:00 PM - 1:30 PM',
      '1:30 PM - 2:00 PM',
      '2:00 PM - 2:30 PM'
    ];
    return (
      <ThrottleCapacityWindow
        showClose
        modalTitle={'test'}
        modalText={'test'}
        strategy={'suggestAndAllow'}
        capacityWindows={capacityWindows}
      />
    );
  }
  )
  .add('Without Keep time slot button', () => {
    let capacityWindows = [
      '1:00 PM - 1:30 PM',
      '1:30 PM - 2:00 PM',
      '2:00 PM - 2:30 PM'
    ];
    return (
      <ThrottleCapacityWindow
        showClose
        modalTitle={'test'}
        modalText={'test'}
        capacityWindows={capacityWindows}
      />
    );
  }
  );
