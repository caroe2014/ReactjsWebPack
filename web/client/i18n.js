// Copyright Â© 2018-2019 Agilysys NV, LLC.  All Rights Reserved.  Confidential Information of Agilysys NV, LLC.

import i18next from 'i18next';
import LanguageDetector from 'i18next-browser-languagedetector';
import Backend from 'i18next-locize-backend';

/* istanbul ignore next */
i18next
  .use(LanguageDetector)
  .use(Backend)
  .init({
    // fallback depending on user language
    fallbackLng: {
      'ja': ['ja-JP'],
      'default': ['en-US']
    },
    order: ['querystring', 'navigator'],
    lookupQuerystring: 'lng',
    allowMultiLoading: true,
    backend: {
      projectId: '838d5fce-27b5-4368-8c54-8fcb33577f9a',
      referenceLng: 'en-US',
      version: process.env.ENVIRONMENT === 'production' ? 'latest' : process.env.ENVIRONMENT || process.env.NODE_ENV
    },
    debug: false,
    detection: {
      caches: []
    },
    // have a common namespace used around the full app
    ns: ['core', `domain-${window.location.hostname}`],
    defaultNS: `domain-${window.location.hostname}`,
    fallbackNS: [`domain-${window.location.hostname}`, 'core'],

    keySeparator: false, // we use content as keys

    interpolation: {
      formatSeparator: ','
    },

    react: {
      wait: true
    }
  }, (err, t) => {
    if (err) return console.log('Something went wrong while loading.', err);
  });

export default i18next;
