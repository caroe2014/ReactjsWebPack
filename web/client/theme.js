// Copyright Â© 2018-2019 Agilysys NV, LLC.  All Rights Reserved.  Confidential Information of Agilysys NV, LLC.

import { css, injectGlobal } from 'styled-components';

const theme = {
  colors: {
    black: '#000000',
    navigationMenu: '#48a942',
    navigationMenuHover: 'cornflowerblue',
    primary: '#347CFF',
    primaryButton: '#347CFF',
    loginField: '#AAA7AC',
    placeHolder: '#909090',
    loginBoxColor: '#434149',
    open: '#48a942',
    close: 'red',
    footer: '#48a942',
    textGrey: '#505050',
    paymentSuccessTitle: '#505050',
    inputBottomBorder: 'lightgray',
    backgroundGrey: '#E6E6E6',
    text: 'black',
    primaryText: '#383838',
    secondaryText: '#a6a6a6',
    hint: '#a6a6a6',
    light: 'white',
    error: '#cc0000',
    tile: '#505050',
    addToCart: '#347CFF',
    itemTile: '#3C474C0',
    darkGrey: '#464646',
    itemBorder: 'lightgrey',
    addOnTab: '#347CFF',
    validationError: '#cc0000',
    loginBackground: '#303030',
    loginFormBackground: '#484848',
    disableBackground: '#E4E4E4',
    addonBg: '#EEEEEE',
    glutenFree: '#8d7356',
    spicy: '#ed1c24',
    vegan: '#0d4a22',
    vegetarian: '#0d4a22',
    new: '#ff6000',
    healthy: '#359919',
    contextBackground: '#f0f0f3',
    loader: '#ababab',
    disableButton: '#D1D1D1'
  },
  fontFamily: 'Roboto',
  receiptFontFamily: 'Roboto',
  fontSize: {
    xl: '2.4rem',
    lg: '1.8rem',
    md: '1.3rem',
    nm: '1rem',
    sm: '0.75rem'
  },
  fontSizeFromProps: (props) => {
    if (props.xl) return props.theme.fontSize.xl;
    if (props.lg) return props.theme.fontSize.lg;
    if (props.md) return props.theme.fontSize.md;
    if (props.nm) return props.theme.fontSize.nm;
    if (props.sm) return props.theme.fontSize.sm;
    return 'inherit';
  },
  spacing: {
    xl: '8em',
    lg: '4em',
    md: '2em',
    nm: '1em',
    sm: '0.5em',
    none: '0em'
  },
  mediaBreakpoints: {
    mobile: (...args) => css`@media (max-width: 32em) {${css(...args)}}`,
    tablet: (...args) => css`@media (max-width: 72em) {${css(...args)}}`,
    desktop: (...args) => css`@media (min-width: 72em) {${css(...args)}}`
  },
  shadow: '#cfcfcf 0.1px 0.1px 3px 0px',
  nav: {
    height: '70px'
  }
};

injectGlobal`
  .agilysys-icon-new {
    color: ${theme.colors.new}
  }
  .agilysys-icon-healthy {
    color: ${theme.colors.healthy}
  }
  .agilysys-icon-vegetarian {
    color: ${theme.colors.vegetarian}
  }
  .agilysys-icon-vegan {
    color: ${theme.colors.vegan}
  }
  .agilysys-icon-spicy {
    color: ${theme.colors.spicy}
  }
  .agilysys-icon-gluten-free {
    color: ${theme.colors.glutenFree}
  }
  @media print {
    @-moz-document url-prefix() {
      .appContainer, .appContainer>div, .appContainer .parent, .appContainer .BottomContainer {
        display: block !important;
      }
    }
  }
`;

export default theme;
