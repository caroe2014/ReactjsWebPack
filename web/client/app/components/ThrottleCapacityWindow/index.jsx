// Copyright Â© 2018-2019 Agilysys NV, LLC.  All Rights Reserved.  Confidential Information of Agilysys NV, LLC.

/* istanbul ignore file */
import React, { Component } from 'react';
import styled from 'styled-components';
import { Modal as ModalWindow, Flex, Fixed, Heading, Text, Button } from 'rebass';
import PrimaryButton from 'web/client/app/components/PrimaryButton';
import RadioButton from 'web/client/app/components/RadioButton';
import IconButton from 'web/client/app/components/IconButton';
import { Trans } from 'react-i18next';
import { scheduleType } from 'web/client/app/utils/constants';
import get from 'lodash.get';
import i18n from 'web/client/i18n';

const ModalContainer = styled(ModalWindow)`
  border-radius: 4px;
  z-index: 1101;
  width: 420px;
  padding: 15px;
  ${props => props.theme.mediaBreakpoints.mobile`width: 100%; border-radius: 0px; height: 100%`};
`;

const ModalBackground = styled(Fixed)`
  background-color: rgba(0, 0, 0, 0.5);
  z-index: 1101;
  top: 0px;
  right: 0px;
  bottom: 0px;
  left: 0px;
  opacity: 1;
  will-change: opacity;
  transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
`;

const ModalTitle = styled(Heading)`
  margin: 25px 20px 25px;
  color: ${props => props.theme.colors.primaryTextColor};
  font-size: 1.5em;
  font-weight: 500;
  text-align: center;
  &:focus {
    outline: none;
    box-shadow: none;
  }
`;

const ModalText = styled(Text)`
  color: ${props => props.theme.colors.secondaryTextColor};
  font-size: 14px;
  text-align: center;
  padding: 0px 20px;
  &:focus {
    outline: none;
    box-shadow: none;
  }
`;

const ModalFooter = styled(Flex)`
  flex-direction: column;
  justify-content: center;
  padding: 20px 20px 0px;
  &:focus {
    outline: none;
    box-shadow: none;
  }
`;

const StyledButton = styled(Button)`
  background-color: ${props => props.theme.colors.buttonControlColor};
  border-radius: 0;
  color: ${props => props.theme.colors.buttonTextColor};
  cursor: pointer;
  font-size: ${props => props.theme.fontSize.md};
  font-weight: 500;
  padding: 20px;
  text-transform: uppercase;
  width: 100%
  &:hover {
    box-shadow: 0px 0px 10px 2px lightgrey;
    -webkit-box-shadow: 0px 0px 10px 2px lightgrey;
    -moz-box-shadow: 0px 0px 10px 2px lightgrey;
  }
`;

const AcceptButton = styled(props => <StyledButton {...props} />)`
  background-color: ${props => props.theme.colors.buttonControlColor};
  opacity: ${props => {
    if (props.selectedoption) return 1;
    return 0.75;
  }};
  border-radius: 0;
  color: ${props => {
    if (props.selectedoption) return props.theme.colors.buttonTextColor;
    return '#FFFFFF';
  }};
  cursor: ${props => {
    if (props.selectedoption) return 'pointer';
    return 'default';
  }};
  font-size: ${props => props.theme.fontSize.md};
  font-weight: 500;
  padding: 20px;
  text-transform: uppercase;
  width: 100%
  &:hover {
    box-shadow: 0px 0px 10px 2px lightgrey;
    -webkit-box-shadow: 0px 0px 10px 2px lightgrey;
    -moz-box-shadow: 0px 0px 10px 2px lightgrey;
  }
  border-radius: 4px;
  font-size: 0.8rem;
  margin-bottom: 20px;
  padding: 15px 0;
  text-align: center;
  width: 100%;
  text-transform: uppercase;
  &:focus {
    outline: none;
    box-shadow: none;
  }
`;

const ContinueButton = styled(props => <PrimaryButton {...props} />)`
  border-radius: 4px;
  font-size: 0.8rem;
  margin-bottom: 20px;
  padding: 15px 0;
  text-align: center;
  width: 100%;
  text-transform: uppercase;
  &:focus {
    outline: none;
    box-shadow: none;
  }
`;

const CloseButton = styled(props => <IconButton {...props} />)`
  color: ${props => props.theme.colors.textGrey};
  font-size: ${props => props.theme.fontSize.md};
  font-weight: 300;
  position: absolute;
  top: 15px;
  &:focus {
    outline: none;
    box-shadow: none;
  }
`;

const RadioButtonContainer = styled(Flex)`
  flex-direction: column;
  width: 100%;
  padding: 25px 20px 0px;
  .input-field-wrapper .input-text {
    border-bottom: 1px solid #7d7d7d;
  }
  .input-field-wrapper {
    margin-bottom: 0px;
  }
  .radio-o {
    font-size: 20px;
    color: ${props => props.theme.colors.buttonControlColor} !important;
  }
`;

class ThrottleCapacityWindow extends Component {
  constructor (props) {
    super(props);
    this.state = {
      selectedOption: null
    };
    this.checkSelected = this.checkSelected.bind(this);
    this.onClickAccept = this.onClickAccept.bind(this);
    this.onClickContinue = this.onClickContinue.bind(this);
    this.onClickOk = this.onClickOk.bind(this);
    this.onClickClose = this.onClickClose.bind(this);
    this.onEscape = this.onEscape.bind(this);
    this.makeEverythingAccessible = this.makeEverythingAccessible.bind(this);
  }

  componentDidMount () {
    document.getElementById('spl-ins-title') &&
      document.getElementById('spl-ins-title').focus();
  }

  checkSelected (type) {
    this.setState({ selectedOption: type });
  }

  onClickAccept () {
    const { selectedOption } = this.state;
    const scheduleSlotList = get(this.props, 'capacityData.scheduleSlotList', []);
    const { scheduleOrderData } = this.props;
    if (selectedOption) {
      const scheduleOrder = {
        scheduleType: scheduleOrderData ? scheduleOrderData.scheduleType : scheduleType.laterToday,
        scheduleTime: selectedOption,
        daysToAdd: scheduleOrderData ? scheduleOrderData.daysToAdd : 0
      };
      this.props.setScheduleOrderData(scheduleOrder);
      this.makeEverythingAccessible();
      this.props.acceptCapacityTime(scheduleSlotList.find(slot => slot.displayRange === selectedOption).scheduledEndTime); // eslint-disable-line max-len
    }
  }

  onClickContinue () {
    this.makeEverythingAccessible();
    this.props.acceptCapacityTime();
  }

  onClickOk () {
    this.makeEverythingAccessible();
    this.props.history.replace('/');
    this.props.cancelCart();
    this.props.resetRemaining();
    this.props.resetDelivery();
    this.props.resetSmsDetails();
    this.props.resetTipData();
    this.props.resetNameDetails();
    this.props.resetScheduleOrderData();
    this.props.setHideScheduleTime(false);
    this.props.clearGAState();
    this.props.resetAtrium();
    this.props.removeCCPaymentCard();
    this.props.clearLoyaltyState();
  }

  onClickClose () {
    const { capacityData, onHandleClose, paymentType } = this.props;
    if (capacityData && capacityData.scheduleSlotList) {
      if (paymentType) {
        this.makeEverythingAccessible();
        onHandleClose && onHandleClose();
      }
      this.makeEverythingAccessible();
      this.props.setShowCapacityWindow(false);
    } else {
      this.onClickOk();
    }
  }

  makeEverythingAccessible () {
    // Accessibility for dialog
    const bottomContainer = document.querySelector('.BottomContainer .pay-list-top-container');
    const topContainer = document.querySelector('.TopContainer');
    topContainer.inert = false;
    const footer = document.querySelector('.BottomContainer .footer');
    footer.inert = false;
    const Children = bottomContainer.children;
    Array.from(Children).forEach(child => {
      if (child.id !== 'throttle-modal') {
        child.inert = false;
      }
    });
  }

  onEscape (e) {
    if (e.which === 27) {
      this.onClickClose();
    }
  }

  render () {
    const { orderThrottling } = this.props;
    const { selectedOption } = this.state;
    const scheduleSlotList = get(this.props, 'capacityData.scheduleSlotList');
    const strategy = get(this.props, 'capacityData.strategy');
    return (
      <Flex id='throttle-modal' role='dialog'>
        <ModalBackground className='throttle-modal-background'/>
        <ModalContainer className='throttle-modal-parent' onKeyDown={this.onEscape}>

          <ModalTitle className='modal-title' tabIndex={0} id='throttle-title'
            aria-label={`${(orderThrottling && orderThrottling.titleText) ||
            i18n.t('THROTTLING_MODAL_TITLE')}`}>
            {(orderThrottling && orderThrottling.titleText) || <Trans i18nKey='THROTTLING_MODAL_TITLE'/>}
          </ModalTitle>
          <ModalText className='modal-text' tabIndex={0}
            aria-label={`${(orderThrottling && (!scheduleSlotList ? orderThrottling.rejectMessage
              : orderThrottling.optionsMessage)) ||
            !scheduleSlotList ? i18n.t('THROTTLING_MODAL_TEXT_REJECT') : i18n.t('THROTTLING_MODAL_TEXT_OPTIONS')}`}
          >
            {(orderThrottling &&
              (!scheduleSlotList ? orderThrottling.rejectMessage : orderThrottling.optionsMessage)) ||
              <Trans i18nKey={!scheduleSlotList ? 'THROTTLING_MODAL_TEXT_REJECT' : 'THROTTLING_MODAL_TEXT_OPTIONS'}/>
            }
          </ModalText>
          {scheduleSlotList && <RadioButtonContainer id='radiogroup' className='throttle-capacity-radio-div'>
            {
              scheduleSlotList.map((capacityRange, index) => (
                <React.Fragment key={`account-${index}`}>
                  <RadioButton
                    classDesc={`${capacityRange.displayRange}`}
                    className={`${capacityRange.displayRange}-radio`}
                    label={capacityRange.displayRange}
                    ariaLabel={capacityRange.displayRange}
                    tabIndex={0}
                    capacityText
                    type={capacityRange.displayRange}
                    selectedOption={this.checkSelected}
                    selected={selectedOption === capacityRange.displayRange}
                  />
                </React.Fragment>
              ))}
          </RadioButtonContainer> }
          <ModalFooter className='modal-footer'>
            {scheduleSlotList && <AcceptButton
              onClick={this.onClickAccept}
              selectedoption={selectedOption}
              role='button'
              tabIndex={selectedOption ? 0 : -1}
              aria-label={i18n.t('THROTTLING_MODAL_ACCEPT')}
            >
              <Trans i18nKey='THROTTLING_MODAL_ACCEPT'/>
            </AcceptButton>}
            {scheduleSlotList && strategy === 'SUGGEST_AND_ALLOW' &&
            <ContinueButton
              role='button'
              tabIndex={0}
              aria-label={i18n.t('THROTTLING_MODAL_KEEP_TIME')}
              onClick={this.onClickContinue}>
              <Trans i18nKey='THROTTLING_MODAL_KEEP_TIME'/>
            </ContinueButton>}
            {!scheduleSlotList &&
              <ContinueButton
                role='button'
                tabIndex={0}
                aria-label={i18n.t('THROTTLING_MODAL_TEXT_REJECT_OK')}
                onClick={this.onClickOk}>
                <Trans i18nKey='THROTTLING_MODAL_TEXT_REJECT_OK'/>
              </ContinueButton>}

          </ModalFooter>
          <CloseButton children='&#10005;' role='button' tabIndex={0}
            aria-label={i18n.t('EXIT_DIALOG')} onClick={this.onClickClose} />
        </ModalContainer>
      </Flex>
    );
  };
}

export default ThrottleCapacityWindow;
