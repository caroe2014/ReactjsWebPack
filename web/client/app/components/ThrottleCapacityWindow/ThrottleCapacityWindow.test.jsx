// Copyright Â© 2018-2019 Agilysys NV, LLC.  All Rights Reserved.  Confidential Information of Agilysys NV, LLC.

import React from 'react';
import { shallow, mount, configure } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import { ThemeProvider } from 'styled-components';
import theme from 'web/client/theme.js';
import ThrottleCapacityWindow from '.';
import jsdom from 'jsdom';

import i18n from 'i18next';

configure({
  adapter: new Adapter()
});

/* global describe, it, expect, beforeEach */
/* eslint-disable max-len */
global.window = new jsdom.JSDOM().window;
global.document = window.document;
global.scrollTo = jest.fn();

jest.mock('web/client/i18n', () => ({
  t: (key) => {
    return key;
  }
}));
jest.mock('react-i18next', () => ({
  // this mock makes sure any components using the translate HoC receive the t function as a prop
  withTranslation: () => Component => {
    Component.defaultProps = { ...Component.defaultProps, t: (key) => key };
    return Component;
  }
}));

jest.mock('web/client/app/reduxpages/ConnectedComponents', () => {
  return {
    ConnectedLoyaltyProcess: <div/>
  };
});

jest.mock('react-i18next', () => ({
  Trans: 'div'
}));

describe('ThrottleCapacityWindow', () => {

  let wrapper;
  let capacityWindows = [
    '1:00 PM - 1:30 PM',
    '1:30 PM - 2:00 PM',
    '2:00 PM - 2:30 PM'
  ];

  beforeEach(() => {
    wrapper = shallow(<ThrottleCapacityWindow
      modalTitle={'test'}
      modalText={'test'}
      capacityData={{
        strategy: 'SUGGEST_AND_ALLOW',
        scheduleSlotList: capacityWindows
      }
      }
    />
    );
  });

  it('does a snapshot test', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render without throwing an error', () => {
    expect(wrapper.exists()).toBe(true);
  });

  it('should mount with themes without throwing an error', () => {
    const mountWrapper = mount(
      <ThemeProvider theme={theme}>
        <ThrottleCapacityWindow
          modalTitle={'test'}
          modalText={'test'}
          capacityData={{
            strategy: 'SUGGEST_AND_ALLOW',
            scheduleSlotList: capacityWindows
          }
          }
        />
      </ThemeProvider>);
    expect(mountWrapper.exists()).toBe(true);
  });

  it('should mount with themes without throwing an error', () => {
    const mountWrapper = mount(
      <ThemeProvider theme={theme}>
        <ThrottleCapacityWindow
          strategy={'SUGGEST_AND_ALLOW'}
          capacityData={{
            strategy: 'SUGGEST_AND_ALLOW',
            scheduleSlotList: capacityWindows
          }
          }
        />
      </ThemeProvider>);
    expect(mountWrapper.exists()).toBe(true);
  });

  it('should mount with orderthrottling text without throwing an error', () => {
    let orderThrottling = {
      titleText: 'Sorry. Our kitchens are super busy right now.',
      optionsMessage: 'Will either of these options work for you?',
      rejectMessage: 'Please pick another time or try again later.'
    };
    const mountWrapper = mount(
      <ThemeProvider theme={theme}>
        <ThrottleCapacityWindow
          orderThrottling={orderThrottling}
          capacityData={{
            strategy: 'SUGGEST_AND_ALLOW',
            scheduleSlotList: capacityWindows
          }
          }
        />
      </ThemeProvider>);
    expect(mountWrapper.exists()).toBe(true);
  });

  it('should mount with orderthrottling text without throwing an error', () => {
    let orderThrottling = {
      titleText: 'Sorry. Our kitchens are super busy right now.',
      optionsMessage: 'Will either of these options work for you?',
      rejectMessage: 'Please pick another time or try again later.'
    };
    const mountWrapper = mount(
      <ThemeProvider theme={theme}>
        <ThrottleCapacityWindow
          orderThrottling={orderThrottling}
          capacityData={{
            strategy: 'SUGGEST_AND_ALLOW',
            scheduleSlotList: capacityWindows
          }
          }
        />
      </ThemeProvider>);
    expect(mountWrapper.exists()).toBe(true);
  });

});
