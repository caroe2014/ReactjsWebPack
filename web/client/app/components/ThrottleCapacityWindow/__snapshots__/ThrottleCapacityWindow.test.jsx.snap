// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThrottleCapacityWindow does a snapshot test 1`] = `
<Flex
  id="throttle-modal"
  role="dialog"
>
  <Styled(Fixed)
    className="throttle-modal-background"
  />
  <Styled(Modal)
    className="throttle-modal-parent"
    onKeyDown={[Function]}
  >
    <Styled(Heading)
      aria-label="THROTTLING_MODAL_TITLE"
      className="modal-title"
      id="throttle-title"
      tabIndex={0}
    >
      <div
        i18nKey="THROTTLING_MODAL_TITLE"
      />
    </Styled(Heading)>
    <Styled(Text)
      aria-label="THROTTLING_MODAL_TEXT_OPTIONS"
      className="modal-text"
      tabIndex={0}
    >
      <div
        i18nKey="THROTTLING_MODAL_TEXT_OPTIONS"
      />
    </Styled(Text)>
    <Styled(Flex)
      className="throttle-capacity-radio-div"
      id="radiogroup"
    >
      <RadioButton
        capacityText={true}
        classDesc="undefined"
        className="undefined-radio"
        selected={false}
        selectedOption={[Function]}
        tabIndex={0}
      />
      <RadioButton
        capacityText={true}
        classDesc="undefined"
        className="undefined-radio"
        selected={false}
        selectedOption={[Function]}
        tabIndex={0}
      />
      <RadioButton
        capacityText={true}
        classDesc="undefined"
        className="undefined-radio"
        selected={false}
        selectedOption={[Function]}
        tabIndex={0}
      />
    </Styled(Flex)>
    <Styled(Flex)
      className="modal-footer"
    >
      <Styled(Component)
        aria-label="THROTTLING_MODAL_ACCEPT"
        onClick={[Function]}
        role="button"
        selectedoption={null}
        tabIndex={-1}
      >
        <div
          i18nKey="THROTTLING_MODAL_ACCEPT"
        />
      </Styled(Component)>
      <Styled(Component)
        aria-label="THROTTLING_MODAL_KEEP_TIME"
        onClick={[Function]}
        role="button"
        tabIndex={0}
      >
        <div
          i18nKey="THROTTLING_MODAL_KEEP_TIME"
        />
      </Styled(Component)>
    </Styled(Flex)>
    <Styled(Component)
      aria-label="EXIT_DIALOG"
      onClick={[Function]}
      role="button"
      tabIndex={0}
    >
      âœ•
    </Styled(Component)>
  </Styled(Modal)>
</Flex>
`;
